Source/SwiftLintFramework/Models/Linter.swift

```diff
@@ -378,7 +381,13 @@ public struct CollectedLinter {
         }
 
         var corrections = [Correction]()
-        for rule in rules.compactMap({ $0 as? any CorrectableRule }) {
+        for rule in rules.compactMap({ (rule: any Rule) -> (any CorrectableRule)? in
```

Comment:
Stylistically, I would prefer a separate `if` statement in the `for` block to check for `shouldRun`. Or you can make use of a `where` clause.

We could even check for `CorrectableRule` in the `for` block to avoid creating a new list and iterating twice.